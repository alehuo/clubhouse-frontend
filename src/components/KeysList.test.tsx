import { Key, KeyType, StudentUnion, User } from "@alehuo/clubhouse-shared";
import { shallow } from "enzyme";
import React from "react";
import KeyList from "./KeysList";

const keys: Key[] = [
  {
    keyId: 1,
    keyType: 1,
    unionId: 1,
    userId: 1,
    created_at: "2019-12-30 12:00:00",
    dateAssigned: "2019-12-30 12:00:00",
    description: "Hello World",
    updated_at: "2019-12-30 12:00:00",
  },
];

const studentUnions: StudentUnion[] = [
  {
    unionId: 1,
    name: "Test Student Union",
    description: "Stdu description",
    created_at: "2019-12-30 12:00:00",
    updated_at: "2019-12-30 12:00:00",
  },
];

const keyTypes: KeyType[] = [
  {
    keyTypeId: 1,
    title: "24hr",
    created_at: "2019-12-30 12:00:00",
    updated_at: "2019-12-30 12:00:00",
  },
];

const users: User[] = [
  {
    firstName: "John",
    lastName: "Doe",
    email: "john@doe.com",
    permissions: 8,
    userId: 1,
    created_at: "2019-12-30 12:00:00",
    updated_at: "2019-12-30 12:00:00",
  },
];

it("KeyList matches snapshot", () => {
  const cmpnt = shallow(
    <KeyList keys={keys} studentUnions={studentUnions} keyTypes={keyTypes} users={users} />,
  );
  expect(cmpnt).toMatchSnapshot();
});
