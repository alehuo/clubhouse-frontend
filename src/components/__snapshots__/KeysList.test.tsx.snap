// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyList matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <KeyList
    keyTypes={
      Array [
        Object {
          "created_at": "2019-12-30 12:00:00",
          "keyTypeId": 1,
          "title": "24hr",
          "updated_at": "2019-12-30 12:00:00",
        },
      ]
    }
    keys={
      Array [
        Object {
          "created_at": "2019-12-30 12:00:00",
          "dateAssigned": "2019-12-30 12:00:00",
          "description": "Hello World",
          "keyId": 1,
          "keyType": 1,
          "unionId": 1,
          "updated_at": "2019-12-30 12:00:00",
          "userId": 1,
        },
      ]
    }
    studentUnions={
      Array [
        Object {
          "created_at": "2019-12-30 12:00:00",
          "description": "Stdu description",
          "name": "Test Student Union",
          "unionId": 1,
          "updated_at": "2019-12-30 12:00:00",
        },
      ]
    }
    users={
      Array [
        Object {
          "created_at": "2019-12-30 12:00:00",
          "email": "john@doe.com",
          "firstName": "John",
          "lastName": "Doe",
          "permissions": 8,
          "updated_at": "2019-12-30 12:00:00",
          "userId": 1,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bordered": true,
      "bsClass": "table",
      "children": Array [
        <thead>
          <tr>
            <th>
              #
            </th>
            <th>
              Name
            </th>
            <th>
              Student union
            </th>
            <th>
              Key type
            </th>
            <th>
              Notes
            </th>
            <th>
              Date assigned
            </th>
          </tr>
        </thead>,
        <tbody>
          <tr>
            <td>
              1
            </td>
            <td>
              John Doe
            </td>
            <td>
              Test Student Union
            </td>
            <td>
              24hr
            </td>
            <td>
              Hello World
            </td>
            <td>
              30.12.2019 12:00
            </td>
          </tr>
        </tbody>,
      ],
      "condensed": true,
      "hover": true,
      "responsive": true,
      "striped": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tr>
            <th>
              #
            </th>
            <th>
              Name
            </th>
            <th>
              Student union
            </th>
            <th>
              Key type
            </th>
            <th>
              Notes
            </th>
            <th>
              Date assigned
            </th>
          </tr>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th>
                #
              </th>,
              <th>
                Name
              </th>,
              <th>
                Student union
              </th>,
              <th>
                Key type
              </th>,
              <th>
                Notes
              </th>,
              <th>
                Date assigned
              </th>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "#",
              },
              "ref": null,
              "rendered": "#",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Name",
              },
              "ref": null,
              "rendered": "Name",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Student union",
              },
              "ref": null,
              "rendered": "Student union",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Key type",
              },
              "ref": null,
              "rendered": "Key type",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Notes",
              },
              "ref": null,
              "rendered": "Notes",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Date assigned",
              },
              "ref": null,
              "rendered": "Date assigned",
              "type": "th",
            },
          ],
          "type": "tr",
        },
        "type": "thead",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <tr>
              <td>
                1
              </td>
              <td>
                John Doe
              </td>
              <td>
                Test Student Union
              </td>
              <td>
                24hr
              </td>
              <td>
                Hello World
              </td>
              <td>
                30.12.2019 12:00
              </td>
            </tr>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <td>
                  1
                </td>,
                <td>
                  John Doe
                </td>,
                <td>
                  Test Student Union
                </td>,
                <td>
                  24hr
                </td>,
                <td>
                  Hello World
                </td>,
                <td>
                  30.12.2019 12:00
                </td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 1,
                },
                "ref": null,
                "rendered": 1,
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "John Doe",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "John Doe",
                ],
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Test Student Union",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Test Student Union",
                ],
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "24hr",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "24hr",
                ],
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Hello World",
                },
                "ref": null,
                "rendered": "Hello World",
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "30.12.2019 12:00",
                },
                "ref": null,
                "rendered": "30.12.2019 12:00",
                "type": "td",
              },
            ],
            "type": "tr",
          },
        ],
        "type": "tbody",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bordered": true,
        "bsClass": "table",
        "children": Array [
          <thead>
            <tr>
              <th>
                #
              </th>
              <th>
                Name
              </th>
              <th>
                Student union
              </th>
              <th>
                Key type
              </th>
              <th>
                Notes
              </th>
              <th>
                Date assigned
              </th>
            </tr>
          </thead>,
          <tbody>
            <tr>
              <td>
                1
              </td>
              <td>
                John Doe
              </td>
              <td>
                Test Student Union
              </td>
              <td>
                24hr
              </td>
              <td>
                Hello World
              </td>
              <td>
                30.12.2019 12:00
              </td>
            </tr>
          </tbody>,
        ],
        "condensed": true,
        "hover": true,
        "responsive": true,
        "striped": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th>
                #
              </th>
              <th>
                Name
              </th>
              <th>
                Student union
              </th>
              <th>
                Key type
              </th>
              <th>
                Notes
              </th>
              <th>
                Date assigned
              </th>
            </tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th>
                  #
                </th>,
                <th>
                  Name
                </th>,
                <th>
                  Student union
                </th>,
                <th>
                  Key type
                </th>,
                <th>
                  Notes
                </th>,
                <th>
                  Date assigned
                </th>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "#",
                },
                "ref": null,
                "rendered": "#",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Name",
                },
                "ref": null,
                "rendered": "Name",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Student union",
                },
                "ref": null,
                "rendered": "Student union",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Key type",
                },
                "ref": null,
                "rendered": "Key type",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Notes",
                },
                "ref": null,
                "rendered": "Notes",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Date assigned",
                },
                "ref": null,
                "rendered": "Date assigned",
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <tr>
                <td>
                  1
                </td>
                <td>
                  John Doe
                </td>
                <td>
                  Test Student Union
                </td>
                <td>
                  24hr
                </td>
                <td>
                  Hello World
                </td>
                <td>
                  30.12.2019 12:00
                </td>
              </tr>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                    1
                  </td>,
                  <td>
                    John Doe
                  </td>,
                  <td>
                    Test Student Union
                  </td>,
                  <td>
                    24hr
                  </td>,
                  <td>
                    Hello World
                  </td>,
                  <td>
                    30.12.2019 12:00
                  </td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 1,
                  },
                  "ref": null,
                  "rendered": 1,
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "John Doe",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "John Doe",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Test Student Union",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Test Student Union",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "24hr",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "24hr",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Hello World",
                  },
                  "ref": null,
                  "rendered": "Hello World",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "30.12.2019 12:00",
                  },
                  "ref": null,
                  "rendered": "30.12.2019 12:00",
                  "type": "td",
                },
              ],
              "type": "tr",
            },
          ],
          "type": "tbody",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`KeyList matches snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <KeyList
    keyTypes={Array []}
    keys={Array []}
    studentUnions={Array []}
    users={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bordered": true,
      "bsClass": "table",
      "children": Array [
        <thead>
          <tr>
            <th>
              #
            </th>
            <th>
              Name
            </th>
            <th>
              Student union
            </th>
            <th>
              Key type
            </th>
            <th>
              Notes
            </th>
            <th>
              Date assigned
            </th>
          </tr>
        </thead>,
        <tbody>
          <tr>
            <td
              colSpan={5}
            >
              No keys.
            </td>
          </tr>
        </tbody>,
      ],
      "condensed": true,
      "hover": true,
      "responsive": true,
      "striped": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tr>
            <th>
              #
            </th>
            <th>
              Name
            </th>
            <th>
              Student union
            </th>
            <th>
              Key type
            </th>
            <th>
              Notes
            </th>
            <th>
              Date assigned
            </th>
          </tr>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th>
                #
              </th>,
              <th>
                Name
              </th>,
              <th>
                Student union
              </th>,
              <th>
                Key type
              </th>,
              <th>
                Notes
              </th>,
              <th>
                Date assigned
              </th>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "#",
              },
              "ref": null,
              "rendered": "#",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Name",
              },
              "ref": null,
              "rendered": "Name",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Student union",
              },
              "ref": null,
              "rendered": "Student union",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Key type",
              },
              "ref": null,
              "rendered": "Key type",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Notes",
              },
              "ref": null,
              "rendered": "Notes",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Date assigned",
              },
              "ref": null,
              "rendered": "Date assigned",
              "type": "th",
            },
          ],
          "type": "tr",
        },
        "type": "thead",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tr>
            <td
              colSpan={5}
            >
              No keys.
            </td>
          </tr>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <td
              colSpan={5}
            >
              No keys.
            </td>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No keys.",
              "colSpan": 5,
            },
            "ref": null,
            "rendered": "No keys.",
            "type": "td",
          },
          "type": "tr",
        },
        "type": "tbody",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bordered": true,
        "bsClass": "table",
        "children": Array [
          <thead>
            <tr>
              <th>
                #
              </th>
              <th>
                Name
              </th>
              <th>
                Student union
              </th>
              <th>
                Key type
              </th>
              <th>
                Notes
              </th>
              <th>
                Date assigned
              </th>
            </tr>
          </thead>,
          <tbody>
            <tr>
              <td
                colSpan={5}
              >
                No keys.
              </td>
            </tr>
          </tbody>,
        ],
        "condensed": true,
        "hover": true,
        "responsive": true,
        "striped": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th>
                #
              </th>
              <th>
                Name
              </th>
              <th>
                Student union
              </th>
              <th>
                Key type
              </th>
              <th>
                Notes
              </th>
              <th>
                Date assigned
              </th>
            </tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th>
                  #
                </th>,
                <th>
                  Name
                </th>,
                <th>
                  Student union
                </th>,
                <th>
                  Key type
                </th>,
                <th>
                  Notes
                </th>,
                <th>
                  Date assigned
                </th>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "#",
                },
                "ref": null,
                "rendered": "#",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Name",
                },
                "ref": null,
                "rendered": "Name",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Student union",
                },
                "ref": null,
                "rendered": "Student union",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Key type",
                },
                "ref": null,
                "rendered": "Key type",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Notes",
                },
                "ref": null,
                "rendered": "Notes",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Date assigned",
                },
                "ref": null,
                "rendered": "Date assigned",
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <td
                colSpan={5}
              >
                No keys.
              </td>
            </tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <td
                colSpan={5}
              >
                No keys.
              </td>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "No keys.",
                "colSpan": 5,
              },
              "ref": null,
              "rendered": "No keys.",
              "type": "td",
            },
            "type": "tr",
          },
          "type": "tbody",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
